
afmformats - 0.18.4
nanite - 4.2.0 

## JPK_reader fails
> [!note] In reply to [afmformats issue#36](https://github.com/AFM-analysis/afmformats/issues/36)

Hi, I think I'm having a similar issue to OP I have a large number of `.force-save` files generated by JPK DP V6 software, none of which can be interpreted by `afmformats.load_data` failing with a similar traceback as shown by PinkShnack:

```
afmformats.errors.FileFormatNotSupportedError: Could not determine file format recipe for <path>!
```

They represent a number of different force indentation experiments some including "Retracted Pause" segments and some omitting them being 3 or 2 segments respectively. When using `afmformats.formats.fmt_jpk.load_jpk`;

Those with 3 segments fail with:
```python
Traceback (most recent call last):
  File "/Workspace/Scripts/test_nanite.py", line 18, in <module>
    indent = afmformats.formats.fmt_jpk.load_jpk(path)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/joe/miniconda3/envs/AFM/lib/python3.12/site-packages/afmformats/formats/fmt_jpk/__init__.py", line 69, in load_jpk
    metadata = jpkr.get_metadata(index=index)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/joe/miniconda3/envs/AFM/lib/python3.12/site-packages/afmformats/formats/fmt_jpk/jpk_reader.py", line 352, in get_metadata
    mdap = copy.deepcopy(self.get_metadata(index, seg))
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/joe/miniconda3/envs/AFM/lib/python3.12/site-packages/afmformats/formats/fmt_jpk/jpk_reader.py", line 394, in get_metadata
    md["imaging mode"] = self.get_imaging_mode()
                         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/joe/miniconda3/envs/AFM/lib/python3.12/site-packages/afmformats/formats/fmt_jpk/jpk_reader.py", line 456, in get_imaging_mode
    raise ValueError("Segment 1 must be of type 'pause'!")
ValueError: Segment 1 must be of type 'pause'!
```

Whereas those with 2 segments fail with:
```python
Traceback (most recent call last):
  File "/Workspace/Scripts/test_nanite.py", line 18, in <module>
    indent = afmformats.formats.fmt_jpk.load_jpk(path)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/joe/miniconda3/envs/AFM/lib/python3.12/site-packages/afmformats/formats/fmt_jpk/__init__.py", line 69, in load_jpk
    metadata = jpkr.get_metadata(index=index)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/joe/miniconda3/envs/AFM/lib/python3.12/site-packages/afmformats/formats/fmt_jpk/jpk_reader.py", line 352, in get_metadata
    mdap = copy.deepcopy(self.get_metadata(index, seg))
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/joe/miniconda3/envs/AFM/lib/python3.12/site-packages/afmformats/formats/fmt_jpk/jpk_reader.py", line 397, in get_metadata
    md_im["position index"])
    ~~~~~^^^^^^^^^^^^^^^^^^
KeyError: 'position index'
```

Indicating that:
- **3-segment files fail** because `afmformats` assumes that the second segment must be a `'pause'` under the assumption that it must be a stress-relaxation experiment. This is a **hardcoded assumption** that is **not valid** with all JPK files.

- **2-segment files fail** because `afmformats` expects metadata keys that aren't getting populated, like `"position index"` which is not present in any of my `.jpk-force` files.
